app/src/db/models/document.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from uuid import UUID
+
+import numpy as np
+from pgvector.sqlalchemy import Vector
+from sqlalchemy import ForeignKey
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from src.db.models.base import Base, IdMixin, TimestampMixin
+
+logger = logging.getLogger(__name__)
+
+
+class Document(Base, IdMixin, TimestampMixin):
+    __tablename__ = "document"
+
+    name: Mapped[str]
+    content: Mapped[str | None]
+
+    chunks: Mapped[list["Chunk"]] = relationship(
+        "Chunk", back_populates="document", cascade="all, delete"
+    )
+
+
+class Chunk(Base, IdMixin, TimestampMixin):
+    __tablename__ = "chunk"
+
+    content: Mapped[str]
+    tokens: Mapped[int | None]
+    embedding: Mapped[np.ndarray] = mapped_column(Vector(768))
```

Comment:
would this cause issues if we were to use a different embedding function?

Reply:
Good spot -- I think there are a couple of options:
 - One is to leave off the length and make it just `mapped_column(Vector)`. Pro of this is we can then insert values with other lengths, but the con is we'll have to make sure we're managing which rows came from which embedding fucntions
  - Another is to rename this column to something more specific, like `mpnet_embedding`, and if we want to switch to different embedding models in the future we add another column
  
  I'm open to both options -- what do you think?

Reply:
I'm partial to option 2. The more I think of it, once we optimize we probably won't be needing to manage multiple embedding functions.

Reply:
👍  just made this change!